config/routes.rb

```diff
@@ -0,0 +1,22 @@
+Rails.application.routes.draw do
+  # get 'users/index'
+  devise_for :users
+  unauthenticated :user do
+    root to: 'users#index'
+  end
+
+  authenticated :user do
+    root to: 'categories#index', as: :authenticated_root
+  end
+
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  resources :categories do
+    resources :transactions
```

Comment:
[REQUIRED] As explained by the previous reviewer using :only and :except options can help to limit our actions to just the needed ones instead of all the seven actions thus leading to more unnecessary memory usage